src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildListener.java

```diff
@@ -64,16 +65,11 @@ public void onCompleted(Run<?, ?> run, @Nonnull TaskListener listener) {
 
     StashRepository repository = trigger.getRepository();
     Result result = run.getResult();
-    // Note: current code should no longer use "new JenkinsLocationConfiguration()"
-    // as only one instance per runtime is really supported by the current core.
-    JenkinsLocationConfiguration globalConfig = JenkinsLocationConfiguration.get();
-    String rootUrl = globalConfig == null ? null : globalConfig.getUrl();
-    String buildUrl = "";
-    if (rootUrl == null) {
-      buildUrl = " PLEASE SET JENKINS ROOT URL FROM GLOBAL CONFIGURATION " + run.getUrl();
-    } else {
-      buildUrl = rootUrl + run.getUrl();
-    }
+
+    final String rootUrl =
+        Objects.toString(
+            Jenkins.getInstance().getRootUrl(), " Jenkins root URL is not configured ");
```

Comment:
this is also in the previous version, but I don't get it: why the leading space in the default value?

Reply:
Apparently for readability. That's how it looks like with the spaces:

**[✓ BUILD SUCCESS]( Jenkins root URL is not configured job/TestRepository_Pipeline_PR_Builder/15/)** - Build &#35;_15_ which took _1.4 sec_

The trailing space is needed to separate "configured" from "job". But the leading space can be removed:

**[✓ BUILD SUCCESS](Jenkins root URL is not configured job/TestRepository_Pipeline_PR_Builder/15/)** - Build &#35;_15_ which took _1.4 sec_

I think the version with the space looks better, but I don't insist. We can also try more "technical" separators for that phrase to indicate that the Jenkins root URL would go there:

**[✓ BUILD SUCCESS](&lt;Jenkins root URL is not configured&gt;/job/TestRepository_Pipeline_PR_Builder/15/)** - Build &#35;_15_ which took _1.4 sec_

Or more visual and less technical:

**[✓ BUILD SUCCESS](*Jenkins root URL is not configured*/job/TestRepository_Pipeline_PR_Builder/15/)** - Build &#35;_15_ which took _1.4 sec_

We can even use a link to a non-existing location. That would be harder to notice, but not so alarming for users who care about build success more that about working links:

**[✓ BUILD SUCCESS](http://jenkins-root-url-is-not-configured/job/TestRepository_Pipeline_PR_Builder/15/)** - Build &#35;_15_ which took _1.4 sec_

What's your preference?

Reply:
Generally I'm for alarming users rather than hiding problems. 
I'm OK with leaving it as is -- was just curious.

Actually maybe you could leave a code comment to explain why the leading space is there?

Reply:
OK, then I'm going to surround the message it with `===` on both sides. It should be clear without comments that it's being done for readability. Also, let's use the original message. Users could google for error messages. No need to throw away the existing knowledge. My real goal here is to minimize the number of conditionals, not to rewrite the message.